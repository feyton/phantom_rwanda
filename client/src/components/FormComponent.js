import React, { useState } from 'react';
import { useForm } from 'react-hook-form';
import { useNavigate } from 'react-router-dom';
import { toast } from 'react-toastify';
import { ButtonA as Button, ButtonLoading } from './Button.js';
import LabelComponent from './LabelComponent.js';

export const FormComponent = ({ type, inputs = [], callback, redirect }) => {
  const navigate = useNavigate();
  const [loading, setloading] = useState(false);
  const {
    register,
    handleSubmit,
    setError,
    clearErrors,
    formState: { errors }
  } = useForm();

  const onValid = async (data) => {
    try {
      setloading(true);
      clearErrors();
      await callback(data);
      navigate(redirect);
    } catch (error) {
      /* istanbul ignore next */
      if (error.response.data) {
        const errorData = error.response.data.data;
        Object.keys(errorData).forEach((key) => {
          setError(key, errorData[key]);
          toast(`${key}:${errorData[key]}`, { type: 'error' });
        });
        return;
      }
      toast(error.message);
    } finally {
      setloading(false);
    }
  };
  const inpuStyles =
    'appearance-none border font-raleway rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden';

  return (
    <div className="py-0  md:flex md:justify-center pt-3">
      <div className="block bg-white rounded  p-2 w-full lg:max-w-3xl shadow-lg min-h-[70vh] mr-4">
        <div className="py-4 px-8 text-black font-bold lg:px-4 md:px-4 ">
          <h1 className="text-center  text-2xl  font-raleway font-bold md:ml-0 md:text-left">
            {' '}
            {type}
          </h1>
        </div>
        <form onSubmit={handleSubmit(onValid)} id="form">
          <div className="py-4 px-4  w-full  lg:mx-auto ">
            {inputs.map((input, index) => {
              if (input.length > 1) {
                return (
                  <div
                    className="block mb-4 xl:flex md:grid md:grid-cols-2 md:gap-2"
                    key={index}
                  >
                    {input.map((field) => {
                      if (field.type === 'select') {
                        return (
                          <div className="mr-1 xl:w-1/2" key={field.name}>
                            <label
                              htmlFor={field.id}
                              name={field.name}
                              className="block text-grey-darker text-sm font-bold mb-2"
                            >
                              {field.labelName}
                            </label>
                            <select
                              id={field.id}
                              placeholder={field.placeholder}
                              className="appearance-none border rounded w-full py-2 px-3 text-grey-darker bg-gray-200"
                              required
                              {...register(field.name, {
                                required: true
                              })}
                            >
                              {field.options.map((option, index) => {
                                return (
                                  <option
                                    key={option.value || option.id}
                                    value={option.value}
                                    className="text-grey-darker"
                                    hidden={index === 0}
                                  >
                                    {' '}
                                    {option.name}{' '}
                                  </option>
                                );
                              })}
                            </select>
                          </div>
                        );
                      }
                      return (
                        <div className="ml-1 xl:w-1/2 " key={field.id}>
                          <div className="mb-2 block font-raleway">
                            <LabelComponent
                              htmlFor={field.id}
                              name={field.labelName}
                            />
                            <input
                              type={field.type || 'text'}
                              name={field.name}
                              className={inpuStyles}
                              placeholder={field.placeholder || field.name}
                              id={field.id}
                              {...register(field.name, {
                                required: `${field.name} is required`,
                                pattern: {
                                  value: field.pattern,
                                  message: `${field.message}`
                                }
                              })}
                            />
                          </div>

                          <p className="text-red-800">
                            {errors && errors[field.name] && field.message}
                          </p>
                        </div>
                      );
                    })}
                  </div>
                );
              }
              const field = input[0];
              return (
                <div className="mb-4" key={field.id}>
                  <div className="w-full mr-3 flex flex-col">
                    <LabelComponent htmlFor={field.id} name={field.labelName} />
                    <input
                      type={field.type || 'text'}
                      name={field.name}
                      className={inpuStyles}
                      placeholder={field.placeholder || field.name}
                      id={field.name}
                      {...register(`${field.name}`, {
                        required: `${field.name} is required`,
                        pattern: {
                          value: field.pattern,
                          message: `${field.message}`
                        }
                      })}
                    />

                    <p className="text-red-800">
                      {errors && errors[field.name] && field.message}
                    </p>
                  </div>
                </div>
              );
            })}

            <div className="flex justify-center align">
              {loading ? (
                <ButtonLoading name="Sending..." />
              ) : (
                <Button name={type} type="submit" id="btn" />
              )}
            </div>
          </div>
        </form>
      </div>
    </div>
  );
};

export default FormComponent;
