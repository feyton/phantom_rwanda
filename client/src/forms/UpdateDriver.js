import React from 'react';
import { useForm } from 'react-hook-form';
import bus2 from '../assets/bus2.jpg';
import Button from '../components/Button.js';
import Input from '../components/Input.js';

const UpdateDriver = () => {
  const {
    register,
    handleSubmit,
    formState: { errors }
  } = useForm();

  return (
    <div>
      <div className="flex lg:mx-24 xl:mx-24 border rounded-xl my-8">
        <div>
          <img
            src={bus2}
            alt="bus"
            className="hidden lg:flex xl:flex h-full rounded-l-xl object-cover"
          />
        </div>
        <div className="mx-12 w-full">
          <h1 className="font-raleway font-bold my-8 text-2xl">
            Update Driver Profile
          </h1>
          <form
            onSubmit={handleSubmit()}
            className="flex flex-col items-center"
          >
            <div className="flex flex-col lg:flex-row xl:flex-row">
              <div className="lg:mr-4 xl:mr-4 lg:w-96 xl:w-96">
                <Input
                  labelName="First Name"
                  placeholder="Enter the first name"
                  type="text"
                  name="firstName"
                  {...register('firstName', {
                    required: 'First name is required',
                    pattern: /^[a-zA-Z][a-zA-Z ]{3,32}$/i
                  })}
                />
                <div className="h-8">
                  {errors.firstName && (
                    <p className="mb-4 error-message required">
                      A valid firstname is required
                    </p>
                  )}
                </div>
                <Input
                  labelName="mobile number"
                  placeholder="Enter the mobile number"
                  type="text"
                  name="mobileNumber"
                  {...register('mobileNumber', {
                    required: 'Mobile number is required',
                    pattern: /(0|7|8)\d{9}$/
                  })}
                />
                <div className="h-8">
                  {errors.mobileNumber && (
                    <p className="mb-4 error-message required">
                      Invalid mobile number
                    </p>
                  )}
                </div>

                <Input
                  labelName="Address"
                  placeholder="Enter the address"
                  type="text"
                  name="address"
                  {...register('address', {
                    required: 'address is required',
                    pattern: /^(?!.* {2})[a-z][a-z ]{3,}/i
                  })}
                />
                <div className="h-8">
                  {errors.address && (
                    <p className="mb-4 error-message required">
                      Invalid address
                    </p>
                  )}
                </div>
              </div>
              <div className="lg:ml-4 xl:ml-4 lg:w-96 xl:w-96">
                <Input
                  labelName="Last name"
                  placeholder="Enter the Last name"
                  type="text"
                  name="lastName"
                  {...register('lastName', {
                    required: 'Last name is required',
                    pattern: /^[a-zA-Z][a-zA-Z ]{3,32}$/i
                  })}
                />
                <div className="h-8">
                  {errors.lastName && (
                    <p className="mb-4 error-message required">
                      A valid last name is required
                    </p>
                  )}
                </div>
                <div className="mb-2 block font-raleway">
                  <label
                    htmlFor="Company"
                    className="block text-grey-darker text-sm font-bold font-rale  mb-2"
                  >
                    Company
                  </label>
                  <select
                    id="company"
                    name="company"
                    {...register('company')}
                    className="border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden mb-1"
                  >
                    <option hidden>Select Company</option>
                    <option value="RFTC">RFTC</option>
                    <option value="KBS">KBS</option>
                    <option value="ROY">Royal</option>
                  </select>
                </div>
                <div className="h-8">
                  {errors.company && (
                    <p className="error-message required">
                      Company name is required
                    </p>
                  )}
                </div>
                <Input
                  labelName="National ID"
                  placeholder="Enter the National ID"
                  type="text"
                  name="nationalID"
                  {...register('nationalID', {
                    required: 'National ID is required',
                    pattern: /^[1-3](19|20)\d{2}[7-8]\d{7}[0-9]\d{2}$/
                  })}
                />
                <div className="h-8">
                  {errors.nationalID && (
                    <p className="error-message required">
                      A valid national ID is required
                    </p>
                  )}
                </div>
              </div>
            </div>
            <Input
              labelName="Driver license"
              placeholder="Enter the Driver license"
              type="text"
              name="driverLicense"
              {...register('driverLicense', {
                required: 'Driver License is required',
                pattern: /^[1-3](19|20)\d{2}[7-8]\d{7}[0-9]\d{2}$/
              })}
            />
            <div className="h-8">
              {errors.driverLicense && (
                <p className="mb-4 error-message required">
                  A valid driver license is required
                </p>
              )}
            </div>
            <Button
              name="Update"
              styles="bg-primary mb-8 text-background py-2"
            />
          </form>
        </div>
      </div>
    </div>
  );
};

export default UpdateDriver;
