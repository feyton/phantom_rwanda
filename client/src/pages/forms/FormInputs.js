export const driverInputs = [
  [
    {
      id: 'firstname',
      pattern: /^[a-zA-Z][a-zA-Z ]{3,32}$/i,
      labelName: 'First name',
      type: 'text',
      name: 'firstName',
      message: 'A valid firstname is required',
      placeholder: 'Vedaste'
    },
    {
      id: 'lastname',
      pattern: /^[a-zA-Z][a-zA-Z ]{3,32}$/i,
      labelName: 'Last name',
      type: 'text',
      name: 'lastName',
      message: 'A valid last name is required',
      placeholder: 'Hategekimana'
    }
  ],
  [
    {
      id: 'email',
      pattern:
        /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/,
      labelName: 'Email',
      type: 'email',
      name: 'email',
      message: 'Invalid email address',
      placeholder: 'example@mail.com'
    },
    {
      id: 'mobileNumber',
      pattern: /(0|7|8)\d{9}$/,
      labelName: 'Mobile number',
      type: 'text',
      name: 'mobileNumber',
      message: 'Invalid mobile number',
      placeholder: '0789000000'
    }
  ],
  [
    {
      type: 'select',
      name: 'company',
      id: 'company',
      placeholder: 'Select company',
      labelName: 'Select company',
      options: [
        { name: 'Select company', id: 'def' },
        {
          value: 'Royal Express Limited',
          name: 'Royal Express Limited',
          id: 'royal'
        },
        { value: 'City Express Ltd', name: 'City Express Ltd', id: 'kbs' },
        { value: 'Kigali bus services', name: 'Kigali bus services', id: 'kbs' }
      ]
    },
    {
      type: 'text',
      name: 'address',
      id: 'address',
      labelName: 'Address',
      pattern: /^(?!.* {2})[a-z][a-z ]{3,}/i,
      message: 'Invalid address',
      placeholder: 'Cell, Sector, District'
    }
  ],
  [
    {
      id: 'nationalid',
      pattern: /^[1-3](19|20)\d{2}[7-8]\d{7}[0-9]\d{2}$/,
      labelName: 'National ID',
      name: 'nationalID',
      message: 'A valid national ID is required',
      placeholder: '1 119950030880 0 40'
    },
    {
      id: 'license',
      pattern: /^[1-3](19|20)\d{2}[7-8]\d{7}[0-9]\d{2}$/,
      labelName: 'Driver License',
      name: 'license',
      message: 'A valid driver license is required',
      placeholder: '1 119950030880 0 40'
    }
  ]
];

export const operatorInputs = driverInputs.slice().fill(
  [
    {
      id: 'nationalid',
      pattern: /^[1-3](19|20)\d{2}[7-8]\d{7}[0-9]\d{2}$/,
      labelName: 'National ID',
      name: 'nationalID',
      message: 'A valid national ID is required',
      placeholder: '1 119950030880 0 40'
    }
  ],
  -1
);
