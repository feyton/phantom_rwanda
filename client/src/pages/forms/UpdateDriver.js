import React from 'react';
import UpdateProfile from './UpdateProfile.js';

const UpdateDriver = () => {
  return (
    <div>
      <UpdateProfile
        addressInfo="Kacyiru, Kigali"
        companyInfo="REL"
        firstNameInfo="Eric"
        lastNameInfo="SHEMA"
        mobileNumberInfo="0789989655"
        nationalIDInfo="1199780020002120"
        driverLicenseInfo="1199780020002120"
        position="Driver"
      />
    </div>
  );
};

export default UpdateDriver;
