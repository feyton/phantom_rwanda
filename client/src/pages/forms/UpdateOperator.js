import React from 'react';
import UpdateProfile from './UpdateProfile.js';

const UpdateOperator = () => {
  return (
    <div>
      <UpdateProfile
        addressInfo="Kacyiru, Kigali"
        companyInfo="KBS"
        firstNameInfo="Patrick"
        lastNameInfo="NGABO"
        mobileNumberInfo="0789989655"
        nationalIDInfo="1199780020002120"
        OperatorLicenseInfo="1199780020002120"
        position="Operator"
        wrapperStyles="hidden"
      />
    </div>
  );
};

export default UpdateOperator;
