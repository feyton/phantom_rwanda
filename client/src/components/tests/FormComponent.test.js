import { FormComponent } from '@components/FormComponent.js';
import { driverInputs } from '@pages/forms/FormInputs.js';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { mount, shallow } from 'enzyme';
import React from 'react';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import UpdateDriver from '../../forms/UpdateDriver.js';
import UpdateOperator from '../../forms/UpdateOperator.js';

describe('Form Component', () => {
  it('Should render a form Componet', () => {
    const elem = renderer
      .create(
        <MemoryRouter>
          <FormComponent />
        </MemoryRouter>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});

describe('Form Components and actions', () => {
  it('should render a Button of Submit', () => {
    const wrapper = mount(
      <MemoryRouter>
        <FormComponent />
      </MemoryRouter>
    );
    expect(wrapper.find('#btn').length).toEqual(1);
  });

  it('should render a Label Component', () => {
    const wrapper = mount(
      <MemoryRouter>
        <FormComponent inputs={driverInputs} />
      </MemoryRouter>
    );
    expect(wrapper.find('LabelComponent').length).toEqual(7);
  });
});

describe('Form Component Action', () => {
  const inputs = [
    [
      {
        id: 'firstname',
        pattern: /^[a-zA-Z][a-zA-Z ]{3,32}$/i,
        labelName: 'First name',
        type: 'text',
        name: 'firstname',
        message: 'A valid firstname is required',
        placeholder: 'Vedaste'
      },
      {
        id: 'lastname',
        pattern: /^[a-zA-Z][a-zA-Z ]{3,32}$/i,
        labelName: 'Last name',
        type: 'text',
        name: 'lastname',
        message: 'A valid lastname is required'
      }
    ]
  ];
  const callbackMock = jest.fn(() => Promise.resolve({ data: {} }));
  beforeEach(() => {
    render(
      <BrowserRouter>
        <FormComponent
          type="Test"
          redirect="/test"
          inputs={inputs}
          callback={callbackMock}
        />
      </BrowserRouter>
    );
  });
  it('Render the details', async () => {
    fireEvent.submit(screen.getByTestId('form'));
    expect(
      await screen.findAllByText(/A valid firstname is required/i)
    ).toHaveLength(1);
  });
  it('Should submit', async () => {
    fireEvent.input(screen.getByTestId('firstname'), {
      target: {
        value: 'Fabrice'
      }
    });
    fireEvent.input(screen.getByTestId('lastname'), {
      target: {
        value: 'Fabrice'
      }
    });
    fireEvent.submit(screen.getByTestId('form'));
    await waitFor(() => {
      expect(
        screen.queryAllByText(/A valid firstname is required/i)
      ).toHaveLength(0);
      expect(callbackMock).toBeCalled();
    });
  });
});

describe('Testingfull form', () => {
  const inputs = [
    [
      {
        id: 'firstname',
        pattern: /^[a-zA-Z][a-zA-Z ]{3,32}$/i,
        labelName: 'First name',
        type: 'text',
        name: 'firstname',
        message: 'A valid firstname is required',
        placeholder: 'Vedaste'
      }
    ],
    [
      {
        id: 'lastname',
        pattern: /^[a-zA-Z][a-zA-Z ]{3,32}$/i,
        labelName: 'First name',
        type: 'text',
        name: 'lastname',
        message: 'A valid lastname is required'
      }
    ]
  ];
  beforeEach(() => {
    render(
      <BrowserRouter>
        <FormComponent
          redirect="/test"
          callback={jest.fn()}
          type="Test"
          inputs={inputs}
        />
      </BrowserRouter>
    );
  });

  it('Should highlight all errors', async () => {
    fireEvent.submit(screen.getByTestId('form'));
    expect(
      await screen.findAllByText(/A valid firstname is required/i)
    ).toHaveLength(1);
  });
});

describe('Render the update', () => {
  it('render the update driver', () => {
    const elem = shallow(<UpdateDriver />);
    expect(elem).toMatchSnapshot();
  });
  it('render the update operator', () => {
    const elem = shallow(<UpdateOperator />);
    expect(elem).toMatchSnapshot();
  });
});
