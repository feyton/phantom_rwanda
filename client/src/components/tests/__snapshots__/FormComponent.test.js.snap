// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component Should render a form Componet 1`] = `
<div
  className="py-0  md:flex md:justify-center pt-3"
>
  <div
    className="block bg-white rounded  p-2 w-full lg:max-w-3xl shadow-lg min-h-[70vh] mr-4"
  >
    <div
      className="py-4 px-8 text-black font-bold lg:px-4 md:px-4 "
    >
      <h1
        className="text-center  text-2xl  font-raleway font-bold md:ml-0 md:text-left"
      >
         
      </h1>
    </div>
    <form
      id="form"
      onSubmit={[Function]}
    >
      <div
        className="py-4 px-4  w-full  lg:mx-auto "
      >
        <div
          className="flex justify-center align"
        >
          <div>
            <button
              className="undefined bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-4 rounded-xl m-2 "
              type="submit"
            />
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`Render the update render the update driver 1`] = `
<div>
  <div
    className="flex lg:mx-24 xl:mx-24 border rounded-xl my-8"
  >
    <div>
      <img
        alt="bus"
        className="hidden lg:flex xl:flex h-full rounded-l-xl object-cover"
        src="bus2.jpg"
      />
    </div>
    <div
      className="mx-12 w-full"
    >
      <h1
        className="font-raleway font-bold my-8 text-2xl"
      >
        Update Driver Profile
      </h1>
      <form
        className="flex flex-col items-center"
        onSubmit={[Function]}
      >
        <div
          className="flex flex-col lg:flex-row xl:flex-row"
        >
          <div
            className="lg:mr-4 xl:mr-4 lg:w-96 xl:w-96"
          >
            <ForwardRef
              labelName="First Name"
              name="firstName"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter the first name"
              type="text"
            />
            <div
              className="h-8"
            />
            <ForwardRef
              labelName="mobile number"
              name="mobileNumber"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter the mobile number"
              type="text"
            />
            <div
              className="h-8"
            />
            <ForwardRef
              labelName="Address"
              name="address"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter the address"
              type="text"
            />
            <div
              className="h-8"
            />
          </div>
          <div
            className="lg:ml-4 xl:ml-4 lg:w-96 xl:w-96"
          >
            <ForwardRef
              labelName="Last name"
              name="lastName"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter the Last name"
              type="text"
            />
            <div
              className="h-8"
            />
            <div
              className="mb-2 block font-raleway"
            >
              <label
                className="block text-grey-darker text-sm font-bold font-rale  mb-2"
                htmlFor="Company"
              >
                Company
              </label>
              <select
                className="border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden mb-1"
                id="company"
                name="company"
                onBlur={[Function]}
                onChange={[Function]}
              >
                <option
                  hidden={true}
                >
                  Select Company
                </option>
                <option
                  value="RFTC"
                >
                  RFTC
                </option>
                <option
                  value="KBS"
                >
                  KBS
                </option>
                <option
                  value="ROY"
                >
                  Royal
                </option>
              </select>
            </div>
            <div
              className="h-8"
            />
            <ForwardRef
              labelName="National ID"
              name="nationalID"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter the National ID"
              type="text"
            />
            <div
              className="h-8"
            />
          </div>
        </div>
        <ForwardRef
          labelName="Driver license"
          name="driverLicense"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Enter the Driver license"
          type="text"
        />
        <div
          className="h-8"
        />
        <Button
          name="Update"
          styles="bg-primary mb-8 text-background py-2"
        />
      </form>
    </div>
  </div>
</div>
`;

exports[`Render the update render the update operator 1`] = `
<div>
  <div
    className="flex lg:mx-24 xl:mx-24 border rounded-xl my-8"
  >
    <div>
      <img
        alt="bus"
        className="hidden lg:flex xl:flex h-full rounded-l-xl object-cover"
        src="bus2.jpg"
      />
    </div>
    <div
      className="mx-12 w-full"
    >
      <h1
        className="font-raleway font-bold my-8 text-2xl"
      >
        Update Operator Profile
      </h1>
      <form
        className="flex flex-col items-center"
        onSubmit={[Function]}
      >
        <div
          className="flex flex-col items-center lg:flex-row xl:flex-row"
        >
          <div
            className="lg:mr-4 xl:mr-4 lg:w-96 xl:w-96"
          >
            <ForwardRef
              labelName="First Name"
              name="firstName"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter the first name"
              type="text"
            />
            <div
              className="h-8"
            />
            <ForwardRef
              labelName="mobile number"
              name="mobileNumber"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter the mobile number"
              type="text"
            />
            <div
              className="h-8"
            />
            <ForwardRef
              labelName="Address"
              name="address"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter the address"
              type="text"
            />
            <div
              className="h-8"
            />
          </div>
          <div
            className="lg:ml-4 xl:ml-4 lg:w-96 xl:w-96"
          >
            <ForwardRef
              labelName="Last name"
              name="lastName"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter the Last name"
              type="text"
            />
            <div
              className="h-8"
            />
            <div
              className="mb-2 block font-raleway"
            >
              <label
                className="block text-grey-darker text-sm font-bold font-rale  mb-2"
                htmlFor="Company"
              >
                Company
              </label>
              <select
                className="border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden mb-1"
                id="company"
                name="company"
                onBlur={[Function]}
                onChange={[Function]}
              >
                <option
                  hidden={true}
                >
                  Select Company
                </option>
                <option
                  value=""
                >
                  RFTC
                </option>
                <option
                  value=""
                >
                  KBS
                </option>
                <option
                  value=""
                >
                  Royal
                </option>
              </select>
            </div>
            <div
              className="h-8"
            />
            <ForwardRef
              labelName="National ID"
              name="nationalID"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter the National ID"
              type="text"
            />
            <div
              className="h-8"
            />
          </div>
        </div>
        <div
          className="h-8"
        />
        <Button
          name="Update"
          styles="bg-primary mb-8 text-background py-2"
        />
      </form>
    </div>
  </div>
</div>
`;
